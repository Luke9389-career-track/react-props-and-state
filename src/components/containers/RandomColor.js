import React, { Component } from 'react';
import ColorDisplay from '../colors/ColorDisplay';

export default class ColorContainer extends Component {
  state = {
    backgroundColor: 'black',
    yuh: false
  };

  setStateColor = () => {
    const colors = ['red', 'green', 'black', 'yellow'];
    const randColor = colors[Math.floor(Math.random() * colors.length)];
    if(randColor === this.state.backgroundColor) {
      this.setState({ backgroundColor: 'transparent', yuh: true });
    } else {
      this.setState({ backgroundColor: randColor, yuh: false });
    }
  }

  componentDidMount() {
    setInterval(() => { this.setStateColor(); }, 1000);
  }

  render() {
    const { backgroundColor, yuh } = this.state;
    return (
      <>
        <ColorDisplay backgroundColor={backgroundColor} yuh={yuh} />
      </>
    );
  }
}
