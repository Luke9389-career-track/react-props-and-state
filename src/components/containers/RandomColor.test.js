import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('Random Color component', () => {
  it('chooses a color', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });

  it('console logs on color click', () => {
    const wrapper = shallow(<RandomColor />);
    const oldColor = wrapper.state('backgroundColor');
    wrapper.instance().setStateColor();
    expect(wrapper.state('backgroundColor') === oldColor).toBe(false) || !expect(wrapper).toMatchSnapshot();
  });
});
