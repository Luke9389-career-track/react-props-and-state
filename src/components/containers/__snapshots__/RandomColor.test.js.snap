// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random Color component chooses a color 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="black"
    yuh={false}
  />
</Fragment>
`;

exports[`Random Color component console logs on color click 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor="yellow"
    yuh={false}
  />
</Fragment>
`;
